### DISCUSSION TITLE
How to switch themes in the admin?

### DISCUSSION THREAD

**Discussion by SequelONE** [Laravel-Backpack/community-forum#618] [Q&A (Help)]
Is there some PHP function to switch themes in the admin? For example, Laravel has a locale entry `App::setLocale('en')`.

**Comment by karandatwani92** [comment]
Hey @SequelONE, you can see how it's working in the demo repo.

- https://github.com/Laravel-Backpack/demo/blob/2834bfb630a2107da117e055da45386d25c62210/routes/backpack/custom.php#L42
- https://github.com/Laravel-Backpack/demo/blob/main/resources/views/vendor/backpack/theme_switch_modal_bs5.blade.php

```php
 Route::post('switch-layout', function (Request $request) {
        $theme = 'backpack.theme-'.$request->get('theme', 'tabler').'::';
        Session::put('backpack.ui.view_namespace', $theme);

        if ($theme === 'backpack.theme-tabler::') {
            Session::put('backpack.theme-tabler.layout', $request->get('layout', 'horizontal'));
        }

        return Redirect::back();
    })->name('tabler.switch.layout');
 ```
 
 ![Screenshot 2023-08-07 at 4 52 40 PM](https://github.com/Laravel-Backpack/community-forum/assets/8214221/0f983b8e-734f-4d40-a0cb-39e16ca4ab84)

**Comment by SequelONE** [comment]
@karandatwani92 thank you. Tried your method. For some reason, themes after the selection do not change. As if the session is recorded at the beginning, and after the redirect is reset. What can be done?
